maps_dashboards/public/components/map_page/map_page.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { i18n } from '@osd/i18n';
+import { AppMountParameters, CoreStart } from 'opensearch-dashboards/public';
+import {
+  NavigationPublicPluginStart as NavigationStart,
+  TopNavMenuData,
+} from '../../../../../src/plugins/navigation/public';
+import { PLUGIN_ID } from '../../../common';
+import { MapContainer } from '../map_container';
+
+interface MapContainerProps {
+  navigation: NavigationStart;
+  setHeaderActionMenu: AppMountParameters['setHeaderActionMenu'];
+  savedObjectsClient: CoreStart['savedObjects']['client'];
+}
+
+export const MapPage = ({
+  navigation,
+  setHeaderActionMenu,
+  savedObjectsClient,
+}: MapContainerProps) => {
+  const topNavMenuConfigs: TopNavMenuData[] = [];
+  topNavMenuConfigs.push({
+    label: i18n.translate('maps.topNav.saveMapButtonLabel', {
+      defaultMessage: `Save`,
+    }),
+    run: async () => {
```

Comment:
use the `showSaveModal` hook from the saved object plugin to show a save modal to add a title and description to your saved object e.g. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/vis_builder/public/application/utils/get_top_nav_config.tsx#L105

This pattern will change a little bit in future so dont build all the other save features into you plugin just yet unless you are okay with some rework, but save and Save As should be okay.

Reply:
Got it! Applied the showSaveModal to the updated commits.